test/functional/pdo_sqlsrv/MsCommon.inc

```diff
@@ -52,6 +53,374 @@ function connect($options=array())
+            if ( stripos( $this->dataType, "char" ) !== false )
+                $append .= "COLLATE Latin1_General_BIN2 ";
+            if ( $this->encType == "deterministic" )
+                $append .= $encDet;
+            else
+                $append .= $encRand;
+        }
+        $append .= $this->options;
+        $colDef = "[" . $this->colName . "] " . $this->dataType . $append;
+        return $colDef;
+    }
+}
+
+
+/**
+ * @return string default column name when a name is not provided in the columnMeta class
+ */
+function get_default_colname( $dataType )
+{
+    $colName = "c_" . str_replace( ",", "_", str_replace( "(", "_", $dataType ));
+    $colName = rtrim( $colName, ")" );
+    return $colName;
+}
+
+
+/**
+ * Create a table
+ * @param object $conn : PDO connection object
+ * @param string $tbname : name of the table to be created
+ * @param array $columnMeta : array of columnMeta objects, which contain metadata for one column
+ */
+function create_table( $conn, $tbname, $columnMeta )
+{
+    try
+    {
+        $dropSql = "IF  EXISTS (SELECT * FROM sys.objects WHERE object_id = OBJECT_ID(N'" . $tbname . "') AND type in (N'U')) DROP TABLE $tbname";
+        $conn->exec( $dropSql );
+        $colDef = "";
+        foreach ( $columnMeta as $meta )
+        {
+            $colDef = $colDef . $meta->getColDef() . ", ";
+        }
+        $colDef = rtrim( $colDef, ", " );
+        $createSql = "CREATE TABLE $tbname ( $colDef )";
+        $conn->exec( $createSql );
+    }
+    catch ( Exception $e )
+    {
+        var_dump( $e );
+        exit;
+    }   
+}
+
+
+/**
+ * class for encapsulating optional parameters for PDOStatement::bindParam
+ */
+class bindParamOption {
+    public $index;      //1-based index of the parameter
+    public $pdoType;    //PDO_PARAM_ type specified for PDOStatement::bindParam
+    public $length;     //length specified for PDOStatement::bindParam
+    public $options;    //options specified for PDOStatement::bindParam
+        
+    function __construct( $index, $pdoType = null, $length = null, $options = null )
+    {
+        $this->index = $index;
+        $this->pdoType = $pdoType;
+        $this->length = $length;
+        $this->options = $options;
+    }
+    /**
+     * @param object $stmt : PDO Statement object
+     * @param mix $var : variable to bind to the SQL statement parameter
+     */
+    function bindWithOp( $stmt, $var )
+    {
+        $stmt->bindParam( $this->index, $var, constant( $this->pdoType ), $this->length, $this->options );
+    }
+}
+
+
+/**
+ * Create a row into a table
+ * @param object $conn : PDO connection object
+ * @param string $tbname : name of the table to be created
+ * @param array $inputs : an associative array column name and its value
+ * @param bool $r : if the row was successfully inserted
+ * $param string $api : PDO_SQLSRV API used for executing the insert query
```

Comment:
Perhaps a more description variable for `$r`, and why is the default value for a boolean flag `null`?

Also, why return `$r` instead of throwing an error right when something wrong happens?

Reply:
Return $r so user can handle the error / continue execution.